pkgs/by-name/tr/treedome/package.nix

```diff
@@ -0,0 +1,134 @@
+{ lib
+, cargo-tauri
+, cmake
+, dbus
+, fetchFromGitea
+, fetchYarnDeps
+, freetype
+, gsettings-desktop-schemas
+, gtk3
+, libsoup
+, mkYarnPackage
+, openssl
+, pkg-config
+, rustPlatform
+, webkitgtk
+, wrapGAppsHook
+, sqlite
+}:
+
+let
+  pname = "treedome";
+  version = "0.3.3";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "solver-orgz";
+    repo = "treedome";
+    rev = "${version}";
+    sha256 = "sha256-492EAKCXyc4s9FvkpqppZ/GllYuYe0YsXgbRl/oQBgE=";
+  };
+
+  frontend-build = mkYarnPackage {
+    inherit version src;
+    pname = "treedome-ui";
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = "${src}/yarn.lock";
+      sha256 = "sha256-rV5jKKnbMutaG5o8gRKgs/uoKwbIkxAPIcx6VWG7mm4=";
+    };
+
+    packageJSON = "${src}/package.json";
```

Comment:
This will cause a build failure when IFD is disabled - I think this is why the CI is currently failing on this PR. I've worked around this before by adding the repo's `package.json` to nixpkgs and pointing it at that - much like you've already done with `Cargo.lock`.

Reply:
You mean yarn.lock?

Reply:
btw, what about the yarn.lock, should I do the same too?

Reply:
Don't need to do it on the `yarnLock`. Only reason `packageJSON` needs it is because `mkYarnPackage` needs to parse the JSON file, so it uses `builtins.readFile` to read it. Since the `package.json` file is a part of the `src` derivation, that results in IFD.